config/spark-defaults.conf

```diff
@@ -1,3 +1,6 @@
+# These are config options shared across ALL Spark nodes and are used
+# when instantiating the node i.e. when running docker compose up -d
+
```

Comment:
I tried to call out which configuration options are doing what, but it's _really_ confusing. There are global options set across all nodes (this file), per worker/master options set by env vars (in compose), and per run options set in `src/submit.sh`.